worker/instancepoller/updater.go

```diff
@@ -189,17 +189,15 @@ func machineLoop(context machineContext, m machine, lifeChanged <-chan struct{},
 	pollInstance := func() error {
 		instInfo, err := pollInstanceInfo(context, m)
 		if err != nil {
-			return err
+			return errors.Trace(err)
 		}
 
 		machineStatus := status.Pending
-		if err == nil {
-			if statusInfo, err := m.Status(); err != nil {
-				logger.Warningf("cannot get current machine status for machine %v: %v", m.Id(), err)
-			} else {
-				// TODO(perrito666) add status validation.
-				machineStatus = status.Status(statusInfo.Status)
-			}
+		if statusInfo, err := m.Status(); err != nil {
```

Comment:
This drive-by is in its own commit. View the fourth commit to see the functional change.